src/styles/points/points.js

```diff
@@ -151,30 +151,42 @@ Object.assign(Points, {
             return;
         }
 
-        // optional sprite
+        // optional sprite and texture
         let sprite_info;
         if (this.hasSprites(style)) {
+            // populate sprite_info object with used sprites
             sprite_info = this.parseSprite(style, draw, context);
+
             if (sprite_info) {
                 style.texcoords = sprite_info.texcoords;
             }
             else {
+                // sprites are defined in the style's texture, but none are used in the current layer
                 return;
             }
+        } else if (draw.sprite) {
+            // sprite specified in the draw layer but no sprites defined in the texture
+            log({ level: 'warn', once: true }, `Layer '${draw.layers[draw.layers.length-1]}': ` +
+            `a sprite '${draw.sprite}' is specified but no sprites are defined in texture '${draw.texture}', skipping features in layer`);
+            return;
         }
 
         // point size defined explicitly, or defaults to sprite size, or generic fallback
         style.size = draw.size;
         if (!style.size) {
+            // a 'size' property has not been set in the draw layer -
+            // use the sprite size if it exists and a generic fallback if it doesn't
             style.size = (sprite_info && sprite_info.css_size) || [DEFAULT_POINT_SIZE, DEFAULT_POINT_SIZE];
         }
         else {
-            style.size = StyleParser.evalCachedPointSizeProperty(draw.size, sprite_info, context);
+            style.size = StyleParser.evalCachedPointSizeProperty(draw.size, sprite_info, Texture.textures[style.texture], context);
             if (style.size == null) {
-                log({ level: 'warn', once: true }, `Layer '${draw.layers[draw.layers.length-1]}': ` +
-                    `'size' includes % and/or ratio-based scaling (${JSON.stringify(draw.size.value)}); ` +
-                    `these can only applied to sprites, but no sprite was specified, skipping features in layer`);
-                return;
+                // the StyleParser couldn't evaluate a sprite size -
+                // this means no sprite was defined, use the texture size
+                let tex = Texture.textures[style.texture];
+                let factor = (draw.size.value.indexOf('%') > -1 ? parseFloat(draw.size.value) : 1.0) / 100.0;
```

Comment:
Wanted to note that the reason for all of the `evalCache...` functions is to avoid (wherever possible) re-evaluating this type of logic per-feature (e.g. if it's a static value like `size: 50%`, it can be evaluated and cached once per texture). While this particular usage (% sizing on texture, without a sprite) is probably not common, the caching can have a significant impact, and it would be better to add this logic to the underlying `StyleParser.evalCachedPointSizeProperty` function (looks like you're passing the full texture in anyway, so should have the context necessary).

Reply:
Ah, I see what happened here – https://github.com/tangrams/tangram/commit/523431ef0ca67a44f15986a4b966b00621d18c1f obviated https://github.com/tangrams/tangram/commit/e3f0d175784436b69c433c7f48106bc0a6e7b9a4 … I'll put a warning back in case something odd happens, but this texture size evaluation is no longer needed.